src/Test/test_nonlinear.jl

```diff
@@ -469,26 +469,29 @@ function test_nonlinear_hs071_NLPBlockDual(model::MOI.ModelLike, config::Config)
         # Get primal solution
         x = MOI.get(model, MOI.VariablePrimal(), v)
         # Get dual solution
-        con_dual = MOI.get(model, MOI.NLPBlockDual())
-        var_dual_lb = MOI.get(model, MOI.ConstraintDual(), cl)
-        var_dual_ub = MOI.get(model, MOI.ConstraintDual(), cu)
-
-        # Evaluate ∇f(x)
-        g = zeros(n_variables)
-        MOI.eval_objective_gradient(evaluator, g, x)
-        # Evaluate ∑ μᵢ * ∇cᵢ(x)
-        jtv = zeros(n_variables)
-        MOI.eval_constraint_jacobian_transpose_product(
-            evaluator,
-            jtv,
-            x,
-            con_dual,
-        )
+        if _supports(config, MOI.BLPBlockDual) &&
```

Comment:
No. `@requires _supports(config, MOI.NLPBlockDual)` at the top of this file.

Reply:
Indeed, there isn't anything left in the test otherwise

Reply:
Yeah all these test functions should now aim to test one thing only. We should skip entire tests, rather than having switches with test parts of tests.

Reply:
But if we do `if MOI.get(model, MOI.DualStatus()) != MOI.NO_SOLUTION`, at least the solver don't have to put anything in `config` since it already implement `DualStatus`

Reply:
I considered this. My thinking that it was better for solvers to explicitly opt-out of attributes that are set by optimize, rather than silently skip ones they should support. 

Solvers not supporting duals have to pass `MOI.ConstraintDual` to exclude, for example.

Reply:
Indeed, that makes sense.